src/dstack/_internal/core/backends/runpod/compute.py

```diff
@@ -0,0 +1,150 @@
+
+    def get_offers(
+        self, requirements: Optional[Requirements] = None
+    ) -> List[InstanceOfferWithAvailability]:
+        offers = get_catalog_offers(
+            backend=BackendType.RUNPOD,
+            requirements=requirements,
+        )
+        offers = [
+            InstanceOfferWithAvailability(
+                **offer.dict(), availability=InstanceAvailability.AVAILABLE
+            )
+            for offer in offers
+        ]
+        return offers
+
+    def run_job(
+        self,
+        run: Run,
+        job: Job,
+        instance_offer: InstanceOfferWithAvailability,
+        project_ssh_public_key: str,
+        project_ssh_private_key: str,
+    ) -> LaunchedInstanceInfo:
+        instance_config = InstanceConfiguration(
+            project_name=run.project_name,
+            instance_name=get_instance_name(run, job),
+            ssh_keys=[
+                SSHKey(public=run.run_spec.ssh_key_pub.strip()),
+                SSHKey(public=project_ssh_public_key.strip()),
+            ],
+            job_docker_config=None,
+            user=run.user,
+        )
+        launched_instance_info = self.create_instance(instance_offer, instance_config)
+        return launched_instance_info
+
+    def create_instance(
+        self,
+        instance_offer: InstanceOfferWithAvailability,
+        instance_config: InstanceConfiguration,
+    ) -> LaunchedInstanceInfo:
+        authorized_keys = instance_config.get_public_keys()
+        memory_size = round(instance_offer.instance.resources.memory_mib / 1024)
+        disk_size = round(instance_offer.instance.resources.disk.size_mib / 1024)
+        try:
+            resp = self.api_client.create_pod(
+                name=instance_config.instance_name,
+                image_name="runpod/pytorch:2.1.1-py3.10-cuda12.1.1-devel-ubuntu22.04",
+                gpu_type_id=instance_offer.instance.name,
+                cloud_type="ALL",  # ["ALL", "COMMUNITY", "SECURE"]:
+                gpu_count=len(instance_offer.instance.resources.gpus),
+                container_disk_in_gb=disk_size,
+                min_vcpu_count=instance_offer.instance.resources.cpus,
+                min_memory_in_gb=memory_size,
+                support_public_ip=True,
+                docker_args=get_docker_args(authorized_keys),
+                ports="22/tcp",
+            )
+        except requests.HTTPError as e:
+            try:
+                details = e.response.json()
+                response_code = details.get("code")
+            except ValueError:
+                raise BackendError(e)
+
+            # code 3: There are no hosts available for your specified virtual machine.
+            if response_code == 3:
+                raise NoCapacityError(details.get("message"))
+            # code 9: Vm cannot be assigned ip from network. Network full
+            if response_code == 9:
+                raise BackendError(details.get("message"))
+
+        instance_id = resp["id"]
```

Comment:
The `create_pod` returns `None` and this line raise TypeError.

For example, if my balance is zero, the 'create_pod' retrun status_code=200 and message will be given.:
```
{"errors":[{"message":"Your account balance is too low to rent a pod. Please add funds to your account.","path":["podFindAndDeployOnDemand"],"extensions":{"code":"RUNPOD"}}],"data":{"podFindAndDeployOnDemand":null}}
```

Reply:
Thanks for pointing it out. I will check for `error` field and` raise BackendError` with messages from runpod.

Reply:
I have updated the PR with changes in Exceptional Handling. I have added exception handling in `terminate_instance` for runpod.